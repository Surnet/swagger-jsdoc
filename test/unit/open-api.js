/* global it, describe, before, beforeEach */

const chai = require('chai');

const { expect } = chai;
const chaiJestSnapshot = require('chai-jest-snapshot');

chai.use(chaiJestSnapshot);

before(() => {
  chaiJestSnapshot.resetSnapshotRegistry();
});

beforeEach(function() {
  chaiJestSnapshot.configureUsingMochaContext(this);
});

describe('OpenAPI specification compatiblity', () => {
  it('The new openapi property is respected', done => {
    // eslint-disable-next-line
    const swaggerJsdoc = require('../../lib');

    const definition = {
      openapi: '3.0.0',
      servers: [
        {
          url: '{scheme}://developer.uspto.gov/ds-api',
          variables: {
            scheme: {
              description: 'The Data Set API is accessible via https and http',
              enum: ['https', 'http'],
              default: 'https',
            },
          },
        },
      ],
      info: {
        description:
          'The Data Set API (DSAPI) allows the public users to discover and search USPTO exported data sets. This is a generic API that allows USPTO users to make any CSV based data files searchable through API. With the help of GET call, it returns the list of data fields that are searchable. With the help of POST call, data can be fetched based on the filters on the field names. Please note that POST call is used to search the actual data. The reason for the POST call is that it allows users to specify any complex search criteria without worry about the GET size limitations as well as encoding of the input parameters.',
        version: '1.0.0',
        title: 'USPTO Data Set API',
        contact: {
          name: 'Open Data Portal',
          url: 'https://developer.uspto.gov',
          email: 'developer@uspto.gov',
        },
      },
    };

    const options = {
      definition,
      apis: [],
    };

    const specification = swaggerJsdoc(options);
    expect(specification).to.matchSnapshot();
    done();
  });
});
