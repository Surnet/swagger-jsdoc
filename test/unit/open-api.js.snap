// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenAPI specification compatiblity The new openapi property is respected 1`] = `
Object {
  "components": Object {},
  "info": Object {
    "contact": Object {
      "email": "developer@uspto.gov",
      "name": "Open Data Portal",
      "url": "https://developer.uspto.gov",
    },
    "description": "The Data Set API (DSAPI) allows the public users to discover and search USPTO exported data sets. This is a generic API that allows USPTO users to make any CSV based data files searchable through API. With the help of GET call, it returns the list of data fields that are searchable. With the help of POST call, data can be fetched based on the filters on the field names. Please note that POST call is used to search the actual data. The reason for the POST call is that it allows users to specify any complex search criteria without worry about the GET size limitations as well as encoding of the input parameters.",
    "title": "USPTO Data Set API",
    "version": "1.0.0",
  },
  "openapi": "3.0.0",
  "paths": Object {},
  "servers": Array [
    Object {
      "url": "{scheme}://developer.uspto.gov/ds-api",
      "variables": Object {
        "scheme": Object {
          "default": "https",
          "description": "The Data Set API is accessible via https and http",
          "enum": Array [
            "https",
            "http",
          ],
        },
      },
    },
  ],
  "tags": Array [],
}
`;
